src/Tools/dotnet-monitor/CollectionRules/Options/CollectionRuleDefaultsPostConfigure/LimitsPostConfigure.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.CollectionRuleDefaultsInterfaces
+{
+    internal sealed class LimitsPostConfigure :
+        IPostConfigureOptions<CollectionRuleOptions>
+    {
+        private readonly IOptionsMonitor<CollectionRuleDefaultsOptions> _defaultOptions;
+
+        public LimitsPostConfigure(IOptionsMonitor<CollectionRuleDefaultsOptions> defaultOptions)
+        {
+            _defaultOptions = defaultOptions;
+        }
+
+        public void PostConfigure(string name, CollectionRuleOptions options)
+        {
+            if (null == options.Limits)
+            {
+                if (!_defaultOptions.CurrentValue.Limits.ActionCount.HasValue
```

Comment:
Rather than constantly accessing `_defaultOptions.CurrentValue.Limits`, cache it in a variable.